File: 3212-count-submatrices-with-equal-frequency-of-x-and-y/3212-count-submatrices-with-equal-frequency-of-x-and-y.py
class Solution:
    def numberOfSubmatrices(self, grid: List[List[str]]) -> int:
        m=len(grid)
        n=len(grid[0])
        sumx=[0]*n
        sumy=[0]*n
        res=0
        for i in range(m):
            rx=0
            ry=0
            for j in range(n):
                if grid[i][j] == 'X':
                    rx += 1
                elif grid[i][j] == 'Y':
                    ry+=1
                sumx[j]+=rx
                sumy[j]+=ry
                if sumx[j]>0 and sumx[j] == sumy[j]:
                    res+=1
        return res
                    

        

File: 3212-count-submatrices-with-equal-frequency-of-x-and-y/3212-count-submatrices-with-equal-frequency-of-x-and-y.java
class Solution {
    public int numberOfSubmatrices(char[][] grid) {
        int rows = grid.length, cols = grid[0].length;
        int[] sumx = new int[cols];
        int[] sumy = new int[cols];
        int res=0;
        for (int i = 0; i<rows;i++){
            int rx=0,ry=0;
            for(int j=0;j<cols;j++){
                if (grid[i][j]=='X') rx++;
                else if (grid[i][j]=='Y')ry++;
                sumx[j]+=rx;
                sumy[j]+=ry;
                if (sumx[j]>0 && sumx[j]==sumy[j]) res++;
            }
            
        }
        return res;
    }
}